src/pages/ThorChainLP/YourPositions.tsx

```diff
@@ -182,87 +66,209 @@ export const YourPositions = () => {
     return allUserLpData.filter(query => query.data?.positions.length)
   }, [allUserLpData])
 
-  // If some are loading, we are loading, but that's not it yet, we also need to check if some are not loaded
-  const someLoading = useMemo(
-    () => allUserLpData.some(position => position.isLoading),
-    [allUserLpData],
-  )
-
   const allLoaded = useMemo(() => {
     return allUserLpData.length && allUserLpData.every(query => query.isSuccess)
   }, [allUserLpData])
 
-  const isEmpty = useMemo(() => allLoaded && !activePositions.length, [allLoaded, activePositions])
-
   const emptyIcon = useMemo(() => <PoolsIcon />, [])
   const headerComponent = useMemo(() => <PoolsHeader />, [])
 
-  const positionRows = useMemo(() => {
-    if (isEmpty) {
-      return (
-        <ResultsEmpty
-          title='pools.yourPositions.emptyTitle'
-          body='pools.yourPositions.emptyBody'
-          icon={emptyIcon}
-        />
-      )
-    }
+  const positions = useMemo(
+    () =>
+      activePositions
+        .flatMap(position => position?.data?.positions ?? [])
+        .filter(isSome)
+        .sort((a, b) => bn(b.totalValueFiatUserCurrency).comparedTo(a.totalValueFiatUserCurrency)),
+    [activePositions],
+  )
 
-    const skeletons = new Array(2).fill(null).map((_, i) => <Skeleton height={16} key={i} />)
+  const columns: Column<UserLpDataPosition>[] = useMemo(
+    () => [
+      {
+        Header: translate('pools.pool'),
+        accessor: 'name',
+        Cell: ({ row }: { value: string; row: RowProps }) => {
```

Comment:
all these cell components could potentially be turned into components so its compartmentalised and not mega file

Reply:
Agree it would make things *some* cleaner though since these aren't reused anywhere, I believe this is fine as-is, this should effectively be thought of as the same `<PositionButton />` we had previously, instead of it being a single component it's a list of smaller cells!